src/jrd/exe_proto.h

```diff
@@ -58,6 +59,29 @@ void EXE_unwind(Jrd::thread_db*, Jrd::Request*);
 
 namespace Jrd
 {
+	class CachedRequestId
+	{
+	public:
+		CachedRequestId()
+			: id(generator++)
+		{
+			fb_assert(id <= MAX_USHORT);
+		}
+
+		CachedRequestId(const CachedRequestId&) = delete;
+		CachedRequestId& operator=(const CachedRequestId&) = delete;
+
+	public:
+		USHORT getId() const
+		{
+			return id;
+		}
+
+	private:
+		unsigned id;
+		static inline std::atomic<unsigned> generator;
```

Comment:
The cache is bound to attachment and not database, what is the point for atomic generator here? BTW, std::atomic_uint is defined.